test/not_null.t.cpp

```diff
@@ -443,7 +482,26 @@ CASE( "not_null<>: Allows to construct from a non-null underlying pointer (uniqu
     EXPECT( &*p == raw );
 }
 
-CASE( "not_null<>: Allows to move from a not-null pointer (unique_ptr)" )
+CASE( "not_null<>: Returns underlying pointer with get() (unique_ptr)" )
+{
+    unique_ptr< int > pi = make_unique< int >(12);
+    int* raw(pi.get());
+    not_null< unique_ptr< int > > p( std::move(pi) );
+
+#if gsl_CONFIG( NOT_NULL_TRANSPARENT_GET )
+    int* pg = p.get();
+    EXPECT( pg == raw );
+#elif gsl_CONFIG( NOT_NULL_GET_BY_CONST_REF )
+    int* pg = p.get().get();
+    EXPECT( pg == raw );
+#else
+    (void) raw;
+    (void) pi;
+    (void) p;
+#endif
+}
```

Comment:
> I still think "transparent" and "constref" should be mutually exclusive, and "transparent" should imply the parts of "constref" that are not the return value of get().

So you're saying `gsl_CONFIG_NOT_NULL_TRANSPARENT_GET=1` should imply `gsl_CONFIG_NOT_NULL_GET_BY_CONST_REF=1`, but explicitly defining `gsl_CONFIG_NOT_NULL_GET_BY_CONST_REF` to any value would then be an error?

Reply:
Agreed.

Reply:
`So you're saying gsl_CONFIG_NOT_NULL_TRANSPARENT_GET=1 should imply gsl_CONFIG_NOT_NULL_GET_BY_CONST_REF=1, but explicitly defining gsl_CONFIG_NOT_NULL_GET_BY_CONST_REF to any value would then be an error?`

In a sense, yes. But it would be more precise to say I think get's return value and the operators' return value should be separated from each other, because when "get is transparent", then "get is returning const ref" is not true, but I think we want the conversion and arrow operators to behave in a more modern way in both cases.

Reply:
Yes, I think we're on the same page now. Please see my latest commit which renders `gsl_CONFIG_NOT_NULL_TRANSPARENT_GET` and `gsl_CONFIG_NOT_NULL_GET_BY_CONST_REF` mutually exclusive.